baybe/constraints/continuous.py

```diff
@@ -31,14 +33,20 @@ class ContinuousLinearInequalityConstraint(ContinuousLinearConstraint):
 
     The constraint is defined as ``sum_i[ x_i * c_i ] >= rhs``, where x_i are the
     parameter names from ``parameters`` and c_i are the entries of ``coefficients``.
-    If you want to implement a constraint of the form `<=`, multiply ``rhs`` and
-    ``coefficients`` by -1. The constraint is typically fulfilled up to a small
-    numerical tolerance.
-
-    The class has no real content as it only serves the purpose of
-    distinguishing the constraints.
+    A constraint of the form `<=` can be achieved by changing the ``operator`` or
+    multiplying ``rhs`` and ``coefficients`` by -1. The constraint is typically
+    fulfilled up to a small numerical tolerance.
     """
 
+    operator: Literal[">=", "<="] = field(default=">=", validator=in_((">=", "<=")))
```

Comment:
Wouldn't this enable us to simply use `ContinuousLinearConstraint` as a single class if we also provide "==" here? This would make the code leaner as we had fewer classes and levels of inheritance. Probably not for this PR as we'd then need to take care of deprecation mechnisms, but would this maybe be something for the next major release?

Reply:
thought about this too, now that you mentioned it as well I've made the corresponding changes. Main reason not to wait is also: were currently doing BayChem upgrades and I dont want to update the respective parts that write the BayBE configs all the time, so I prefer this change din one go

There is one thing I didnt change yet: `SubspaceContinuous` now in principle also only needs `constraints_lin` and not the two current lists `constraints_lin_eq` and `constraints_lin_ineq`. However, this could be seen as a somewhat breaking change. One could cirument it via some work with converting deprecated args etc, but Im not a fan of that as all of this has to be deconstructed again anyway and the plain constructors are not super user-facing... So I favor it as a breaking change. The latter however, should only be done for the next major version -> hence I left the existing lists in `SubspaceContinuous`

The change resulted in quite a lot more of changed files and lines, please have another look and resolve this here if you approve

Reply:
Done, just had three minor comments.